app/controllers/spree/adyen_redirect_controller.rb

```diff
@@ -67,6 +64,19 @@ def payment_method
       @payment_method ||= Gateway::AdyenHPP.last # find(params[:merchantReturnData])
     end
 
+    def current_order
+      @__adyen_current_order ||=
+        Spree::Order.incomplete.find_by!(current_order_params)
+    end
+
+    def current_order_params
+      { number: params[:merchantReference],
+        store_id: current_store.id,
+        user_id: try_spree_current_user.try(:id),
+        currency: current_currency
```

Comment:
`current_store` is set by 
```
  def store_key
    @request.headers['HTTP_SPREE_STORE'] || @request.env['SERVER_NAME']
  end
```
So it will be correct as long as they come back to the proper host. `current_currency` comes from
```
def current_currency
  Spree::Config[:currency]
end
```
I suppose current_user could potential still affect this, as warden will authenticate by looking in the session. 

I'll remove params other that `number`, but this is still going to have issues if current_user is not set. 

Let's say that the session was cleared and the user has not been loaded. We look up the order, and complete it. we then reach line 49 and redirect to the order checkout controller.
Once we 
https://github.com/solidusio/solidus/blob/master/frontend/app/controllers/spree/checkout_controller.rb#L17